static/js/compose_actions.js

```diff
@@ -445,6 +443,8 @@ exports.quote_and_reply = function (opts) {
         content += `${fence}quote\n${message.raw_content}\n${fence}`;
         compose_ui.replace_syntax("[Quoting…]", content, textarea);
         autosize.update($("#compose-textarea"));
+        // Update textarea caret to point to the new line after quoted message.
+        textarea.caret(prev_caret + content.length + 1);
```

Comment:
I'm suspicious of adding `1` here -- isn't that only correct when `prev_caret` was 0?

Reply:
No, +1 is for the `\n` that is being added after the content in this line:
```
 compose_ui.insert_syntax_and_focus("[Quoting…]\n", textarea);
```
as the `replace_content` function only replaces `[Quoting...]` and so the `\n` is not counted in `content.length`.

Reply:
Wait, so is this a separate pre-existing bug? If so, can you fix it in a separate commit with a commit message explaining the issue? It's really helpful to not combine changes for making reviewers able to efficiently understand why a set of changes is correct.

Reply:
Sure, will do.

Reply:
Reflecting on this, I don't think it is a bug. The purpose of adding a newline after `[Quoting…]` is that while quoting a message, the `[Quoting…]` text is immediately added to the compose-box while the insertion of the actual quoted message may take time (depending on internet speed) so it provides a feedback to the user that the quote is in process of being added. 

So, not adding a trailing newline here would mix up this feedback with the rest of the message.